### PR TITLE
Update source-controller to v0.24.0

### PR DESCRIPTION
**Pull request by pjbgf**

### CODE REVIEW

**Comment by hiddeco** [comment]
This is missing the updates to `api/`.

File: Makefile [hunk 1]
```diff
@@ -20,12 +20,12 @@ BUILD_PLATFORMS ?= linux/amd64,linux/arm64,linux/arm/v7
 CACHE := cache
 
 # Version of the source-controller from which to get the GitRepository CRD.
-# Change this if you bump the source-controller/api version in go.mod.
-SOURCE_VER ?= v0.22.5
+# Pulls source-controller/api's version set in go.mod.
+SOURCE_VER ?= $(shell go list -m all | grep github.com/fluxcd/source-controller/api | awk '{print $$2}')
```
Comment (makkes) [inline Makefile:24]:
```suggestion
SOURCE_VER ?= $(shell go list -m github.com/fluxcd/source-controller/api | awk '{print $$2}')
```

File: Makefile [hunk 2]
```diff
@@ -20,12 +20,12 @@ BUILD_PLATFORMS ?= linux/amd64,linux/arm64,linux/arm/v7
...
 
 # Version of the image-reflector-controller from which to get the ImagePolicy CRD.
-# Change this if you bump the image-reflector-controller/api version in go.mod.
-REFLECTOR_VER ?= v0.17.1
+# Pulls image-reflector-controller/api's version set in go.mod.
+REFLECTOR_VER ?= $(shell go list -m all | grep github.com/fluxcd/image-reflector-controller/api | awk '{print $$2}')
```
Comment (makkes) [inline Makefile:28]:
```suggestion
REFLECTOR_VER ?= $(shell go list -m github.com/fluxcd/image-reflector-controller/api | awk '{print $$2}')
```

File: Makefile (hunk 1 above)
Comment (pjbgf) [reply Makefile]: Nice one, updated it accordingly.

**Review by makkes — APPROVED** [review]

**Review by hiddeco — APPROVED** [review]

**Pull request merged**